tidy3d/components/heat_charge/monitor_data.py

```diff
@@ -235,4 +242,220 @@ def symmetry_expanded_copy(self) -> VoltageData:
         return self.updated_copy(voltage=new_phi, symmetry=(0, 0, 0))
 
 
-HeatChargeMonitorDataType = Union[TemperatureData, VoltageData]
+class HeatChargeDataset(Tidy3dBaseModel):
+    """Class that deals with parameter-depending fields."""
+
+    base_data: FieldDataset = pd.Field(title="Base data", description="Spatial dataset")
+
+    field_series: Dict[str, Tuple[IndexedDataArray, ...]] = pd.Field(
+        title="Field series", description="Dictionary of field solutions. "
+    )
+
+    doping: Dict[str, IndexedDataArray] = pd.Field(
+        title="Doping", description="Dictionary containing the doping distributions."
+    )
+
+    parameter_array: Tuple[pd.FiniteFloat, ...] = pd.Field(
+        title="Parameter array",
+        description="Array containing the parameter values at which the field series are stored.",
+    )
+
+    @cached_property
+    def num_fields_saved(self):
+        """Number of fields stored"""
+        return len(self.parameter_array)
+
+    def get_field(self, field: str, loc: int):
+        """Returns the field specified by 'field' stored at the position specified by 'loc'"""
+
+        assert loc < self.num_fields_saved
+        return self.base_data.updated_copy(values=self.field_series[field][loc])
+
+    def get_doping(self, doping_type: str = "Acceptors"):
+        """Returns the doping specified by 'doping_type'. This, can take the values:
+        'Acceptors' or 'Donors'"""
+
+        return self.base_data.updated_copy(values=self.doping[doping_type])
+
+    @pd.root_validator(skip_on_failure=True)
+    def check_data_and_params_have_same_length(cls, values):
+        """Check that both field series and parameter array have the same length."""
+
+        field_series = values["field_series"]
+        parameter_array = values["parameter_array"]
+        for key in field_series.keys():
+            assert len(field_series[key]) == len(parameter_array)
+
+        return values
+
+
+class ChargeSimulationData(HeatChargeMonitorData):
```

Comment:
Do we need a separate `ChargeSimulationData` in addition to `HeatChargeSimulationData`? or did you mean this to be more of `ChargeSimulationMonitorData`/`ChargeMonitorData`? In that case it shouldn't have field `monitor`

Reply:
sorry, ignore this comment, misread the class definition